src/Image.zig

```diff
@@ -153,50 +139,32 @@ pub fn create(allocator: Allocator, width: usize, height: usize, pixel_format: P
         .allocator = allocator,
         .width = width,
         .height = height,
-        .data = .{
-            .image = try PixelStorage.init(allocator, pixel_format, width * height),
-        },
+        .pixels = try PixelStorage.init(allocator, pixel_format, width * height),
     };
 
     return result;
 }
 
 /// Return the pixel format of the image
-pub fn pixelFormat(self: Self) ?PixelFormat {
-    return switch (self.data) {
-        .empty => {
-            return null;
-        },
-        .image => |image| {
-            return std.meta.activeTag(image);
-        },
-        .animation => |animation| {
-            if (animation.frames.items.len > 0) {
-                return std.meta.activeTag(animation.frames.items[0].pixels);
-            }
+pub fn pixelFormat(self: Self) PixelFormat {
+    if (self.animation) |animation| {
```

Comment:
Didn't we agree that first frame will be just a copy of pixels and that pixels will be set even when there is an animation? In that case this and similar methods could just operate on pixels and this if would not be needed.